### PR TITLE
Replace `/` division with multiplication and custom `divide()` function

### LINKED ISSUE #34051: Sass deprecation warning: Using `/` for division is deprecated
dart-sass has deprecated `/` in favor of `math.div`. Compiling https://github.com/twbs/bootstrap/blob/main/scss/_variables.scss now results with many sass warnings.

```
368 │ $navbar-padding-horizontal:        floor(($grid-gutter-width / 2)) !default;
    │                                           ^^^^^^^^^^^^^^^^^^^^^^
    ╵
    node_modules/bootstrap-sass/assets/stylesheets/bootstrap/_variables.scss 368:43  @import

DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($grid-gutter-width, 2)

More info and automated migrator: https://sass-lang.com/d/slash-div
```

### PR DESCRIPTION
**Pull request by mdo**
Fixes #34051.

This PR aims to resolve the Sass `/` deprecation issues from Dart Sass. One direction could've been to drop Libsass entirely, using the Dart Sass math module, but that'd effectively ruin any hope for support for Libsass-based projects. That matters not because we want to support something old and deprecated, but because there are dozens if not hundreds of projects that haven't updated to Dart Sass yet. Plus, it's easy enough for us to resolve on our own end without breaking things for others, so why not.

To make this happen, I've replaced the bulk of the division with multiplication, and then used a custom `divide()` function (thanks @alpadev!) to replace what I couldn't easily switch to multiplication.

Once we get this reviewed and merged, we'll be able to ship v5.0.2.

/cc @twbs/css-review

### CODE REVIEW

File: scss/_functions.scss [hunk 1]
```diff
@@ -95,7 +95,7 @@
 // Color contrast
 // See https://github.com/twbs/bootstrap/pull/30168
 
-// A list of pre-calculated numbers of pow(($value / 255 + .055) / 1.055, 2.4). (from 0 to 255)
+// A list of pre-calculated numbers of pow(($value * .555 + .055) / 1.055, 2.4). (from 0 to 255)
 // stylelint-disable-next-line scss/dollar-variable-default, scss/dollar-variable-pattern
```
Comment (ceedriic) [inline scss/_functions.scss:99]: This is wrong. / 255 is not the same as *.555. It should be * .00392

File: scss/_functions.scss [hunk 2]
```diff
@@ -137,7 +137,7 @@ $_luminance-list: .0008 .001 .0011 .0013 .0015 .0017 .002 .0022 .0025 .0027 .003
   );
 
   @each $name, $value in $rgb {
-    $value: if($value / 255 < .03928, $value / 255 / 12.92, nth($_luminance-list, $value + 1));
+    $value: if($value * .555 < .03928, divide($value * .555, 12.92), nth($_luminance-list, $value + 1));
     $rgb: map-merge($rgb, ($name: $value));
```
Comment (ceedriic) [inline scss/_functions.scss:141]: Same issue as above. In addition, the divide() function could be removed completely (1/255/12.92) = * .0003035

File: scss/_variables.scss [hunk 3]
```diff
@@ -405,7 +405,7 @@ $aspect-ratios: (
   "1x1": 100%,
   "4x3": calc(3 / 4 * 100%),
   "16x9": calc(9 / 16 * 100%),
-  "21x9": calc(9 / 21 * 100%)
+  "21x9": calc(9 * .51 * 100%)
```
Comment (tagliala) [inline scss/_variables.scss:408]: I think this is a find & replace mistake and `9 / 21` should not be converted

**Comment by shehi** [comment]
I don't think these "division-to-multiplication" conversions across files should be made, as it adds complexity to reading the whole code. Why not use newly introduced `divide()` mixin instead? Cognitively, seeing something divided by 2, or multiplied by 0.5 doesn't lead one to the same reasoning immediately, which might make things more brain numbing. I say all this as someone who checks these values regularly, overriding many of them for my purposes.

File: scss/_variables.scss (hunk 3 above)
Comment (mmarton) [reply scss/_variables.scss:408]: looks like `/ 2`1 got replaced by `* .5`1

File: scss/_functions.scss (hunk 2 above)
Comment (voltaek) [reply scss/_functions.scss:141]: I think moving away from 255 - a recognizable number with obvious context when dealing with RGB - to some magic numbers is a bad idea. I'd argue that the reason this dividing twice wasn't previously simplified was for the same reason: to keep the 255. If we have a divide function, we should use it and preserve the readability and ease of maintenance.

Comment (ceedriic) [reply scss/_functions.scss:141]: Or a little comment should be added that states : 0.0003035 = 1/255/12.92

File: scss/_functions.scss (hunk 1 above)
Comment (crifpeguy) [reply scss/_functions.scss:99]: > looks like `/ 2`1 got replaced by `* .5`1
I think the same thing happened here.

File: scss/_functions.scss (hunk 2 above)
Comment (ceedriic) [reply scss/_functions.scss:141]: And to be honest, maybe the .03928 and 12.92 values also deserve a comment ;)

File: scss/_functions.scss [hunk 4]
```diff
@@ -95,7 +95,7 @@
 // Color contrast
 // See https://github.com/twbs/bootstrap/pull/30168
 
-// A list of pre-calculated numbers of pow(($value / 255 + .055) / 1.055, 2.4). (from 0 to 255)
+// A list of pre-calculated numbers of pow(($value * .555 + .055) / 1.055, 2.4). (from 0 to 255)
```
Comment (mdo) [inline scss/_functions.scss:98]:
```suggestion
// A list of pre-calculated numbers of pow((divide($value / 255) + .055) / 1.055, 2.4). (from 0 to 255)
```

File: scss/_functions.scss [hunk 5]
```diff
@@ -137,7 +137,7 @@ $_luminance-list: .0008 .001 .0011 .0013 .0015 .0017 .002 .0022 .0025 .0027 .003
   );
 
   @each $name, $value in $rgb {
-    $value: if($value / 255 < .03928, $value / 255 / 12.92, nth($_luminance-list, $value + 1));
+    $value: if($value * .555 < .03928, divide($value * .555, 12.92), nth($_luminance-list, $value + 1));
```
Comment (mdo) [inline scss/_functions.scss:140]:
```suggestion
    $value: if(divide($value / 255) < .03928, divide($value / 255, 12.92), nth($_luminance-list, $value + 1));
```

File: scss/_variables.scss (hunk 3 above)
Comment (mdo) [inline scss/_variables.scss:408]:
```suggestion
  "21x9": calc(9 / 21 * 100%)
```

File: scss/_functions.scss [hunk 6]
```diff
@@ -95,7 +95,7 @@
 // Color contrast
 // See https://github.com/twbs/bootstrap/pull/30168
 
-// A list of pre-calculated numbers of pow(($value * .555 + .055) / 1.055, 2.4). (from 0 to 255)
+// A list of pre-calculated numbers of pow((divide($value / 255) + .055) / 1.055, 2.4). (from 0 to 255)
```
Comment (voltaek) [inline scss/_functions.scss:98]: Needs comma, not a forward slash.

File: scss/_functions.scss [hunk 7]
```diff
@@ -137,7 +137,7 @@ $_luminance-list: .0008 .001 .0011 .0013 .0015 .0017 .002 .0022 .0025 .0027 .003
   );
 
   @each $name, $value in $rgb {
-    $value: if($value * .555 < .03928, divide($value * .555, 12.92), nth($_luminance-list, $value + 1));
+    $value: if(divide($value / 255) < .03928, divide($value / 255, 12.92), nth($_luminance-list, $value + 1));
```
Comment (voltaek) [inline scss/_functions.scss:140]: Needs comma, not a forward slash.

**Review by voltaek — CHANGES_REQUESTED** [review]
Looks like some copy/paste errors

**Comment by mdo** [comment]
> I don't think these "division-to-multiplication" conversions across all files should be made, as it adds complexity to reading the whole code. Why not use newly introduced `divide()` mixin instead?

I didn't want to require a function where it wasn't explicitly needed. Dividing by 2 or multiplying by .5 feels common enough to me to avoid changing `$value / 2` to `divide($value, 2)`, which is longer and inherently more complicated than `$value * .5`.

**Comment by shehi** [comment]
> > I don't think these "division-to-multiplication" conversions across all files should be made, as it adds complexity to reading the whole code. Why not use newly introduced `divide()` mixin instead?
> 
> I didn't want to require a function where it wasn't explicitly needed. Dividing by 2 or multiplying by .5 feels common enough to me to avoid changing `$value / 2` to `divide($value, 2)`, which is longer and inherently more complicated (because of the function) than `$value * .5`.

Agreed for simple divisions. And I see you fixed this issue for more complicated divisions. All is good, thank you!

**Comment by kelunik** [comment]
Do you plan to backport the same thing to 4.x?

**Comment by AdamGaskins** [comment]
> Do you plan to backport the same thing to 4.x?

Here's the PR: #34571 looks like we're close! :)

**Pull request merged**